pkgs/by-name/ac/acquire/package.nix

```
@@ -0,0 +1 @@
+{ python3Packages }: with python3Packages; toPythonApplication acquire
```

Every such migration breaks the override interface for these packages - you won't be able to override anything else than `python3Packages` anymore. We are working on improvements for this, but I think at this time such a simple migration of these to by-name is not possible.

---

(you lost the `toPythonApplication` along the way, but that's a minor detail)

You can probably simplify with `intersectAttrs`?

There's a problem with that approach, though: You won't get any error anymore when passing arguments that are not part of the package's override interface.

Aka:

```
nix-repl> hello.override { foo = "bar"; }
       error: function 'anonymous lambda' called with unexpected argument 'foo'
```

This will just silently ignored `foo`, right?

---

It can be simplified to at least

```nix
{ lib, python3Packages, ... }@args:

let
  inherit (python3Packages.somePkg) override;
in

python3Packages.toPythonApplication (
  override (
    removeAttrs args [
      "lib"
      "python3Packages"
    ]
    // lib.intersectAttrs (lib.functionArgs override) args
  )
)
```

…which could potentially be abstracted into e.g.

```nix
{ python3Packages, ... }@args: python3Packages.applicationPackage python3Packages.somePkg args
```

at the risk of breaking if you add more arguments (e.g. for default overrides), or something like

```nix
let
  f =
    { python3Packages, ... }@args: python3Packages.applicationPackage python3Packages.somePkg f args;
in
f
```

But I agree that none of these are satisfying; this is just working around the `by-name` override interface not being great, which we should fix and which I am currently in the “thinking really hard about the problem” stage of trying to address.

---

> `lib.intersectAttrs (lib.functionArgs override) args`

This suffers from https://github.com/NixOS/nixpkgs/pull/454559#discussion_r2480385902, too, right?